apps/prairielearn/src/ee/lib/lti13.ts

```diff
@@ -588,29 +605,43 @@ export async function fetchRetry(
   try {
     const response = await fetch(input, opts);
 
-    switch (response.status) {
-      // 422 Unprocessable Entity, User not found in course or is not a student
-      case 422:
-        return response;
-    }
-
     if (!response.ok) {
-      throw new AugmentedError('LTI 1.3 fetch error, please try again', {
+      const resObject = (await response.json()) ?? {};
+      const resString = JSON.stringify(resObject);
```

Comment:
Parsing the response as JSON and then immediately stringifying it is both unnecessary and unsafe. Canvas will send a plaintext response if a rate limit is exceeded:

https://github.com/instructure/canvas-lms/blob/18aaff876882b23e2e410228febe78c3c7a91274/app/middleware/request_throttle.rb#L254-L258

That will cause `response.json()` to fail. You should use `response.text()` here instead.

Reply:
node-fetch won't let you parse the response twice. From their documentation, "Per the specs, a consumed body cannot be used again." So if we get it as an object, we need to manipulate it.

Reply:
Refactored in c61b6c1c25